services/AttendanceService.ts

```diff
@@ -37,21 +39,43 @@ export default class AttendanceService {
       if (!event) throw new NotFoundError('Oh no! That code didn\'t work.');
       if (!event.isOngoing()) throw new UserError('You can only enter the attendance code during the event!');
 
-      const attendanceRepository = Repositories.attendance(txn);
-      const hasAlreadyAttended = await attendanceRepository.hasUserAttendedEvent(user, event);
+      const hasAlreadyAttended = await Repositories.attendance(txn).hasUserAttendedEvent(user, event);
       if (hasAlreadyAttended) throw new UserError('You have already attended this event');
 
-      const attendedAsStaff = asStaff && user.isStaff() && event.requiresStaff;
-      const pointsEarned = attendedAsStaff ? event.pointValue + event.staffPointBonus : event.pointValue;
-      const activityRepository = Repositories.activity(txn);
-      await activityRepository.logActivity(
-        user,
-        attendedAsStaff ? ActivityType.ATTEND_EVENT_AS_STAFF : ActivityType.ATTEND_EVENT,
-        pointsEarned,
-      );
-      await Repositories.user(txn).addPoints(user, pointsEarned);
+      const attendance = await this.writeEventAttendance(user, event, asStaff, txn);
+      return attendance.getPublicAttendance();
+    });
+  }
+
+  private async writeEventAttendance(user: UserModel, event: EventModel, asStaff: boolean,
+    txn: EntityManager): Promise<AttendanceModel> {
+    const attendedAsStaff = asStaff && user.isStaff() && event.requiresStaff;
+    const pointsEarned = attendedAsStaff ? event.pointValue + event.staffPointBonus : event.pointValue;
+
+    await Repositories.activity(txn).logActivity(
+      user,
+      attendedAsStaff ? ActivityType.ATTEND_EVENT_AS_STAFF : ActivityType.ATTEND_EVENT,
+      pointsEarned,
+    );
+    await Repositories.user(txn).addPoints(user, pointsEarned);
+
+    const attendance = await Repositories.attendance(txn).attendEvent(user, event, attendedAsStaff);
+    return attendance;
+  }
+
+  public async submitAttendanceForUser(userUuid: Uuid, eventUuid: Uuid, asStaff = false): Promise<PublicAttendance> {
```

Comment:
Do you think it'd be worthwhile to track somewhere that this attendance was made by another user? e.g. in the Activity comment or something

Reply:
Yeah good idea, I'll add a short activity comment for those attendances